excel-export-service/src/errors.ts

```diff
@@ -0,0 +1,39 @@
+import { NextFunction } from "express";
+import { CustomExpressRequest, CustomExpressResponse } from "./types";
+
+/**
+ * * @extends Error
+ */
+export class ExtendableError extends Error {
+  status: number;
+  isPublic: boolean;
+  originalMessage: string | null;
+
+  constructor(message: string, status: number, isPublic: boolean, originalMessage: string | null) {
+    super(message);
+    this.name = this.constructor.name;
+    this.message = message;
+    this.status = status;
+    this.isPublic = isPublic;
+    this.originalMessage = originalMessage;
+  }
+}
+
+/**
+ * Executes callback function and if error occures, it sends a notification email.
```

Comment:
what notification email?  *confusion*

Reply:
@SamuelPull Sorry, my bad. I used the structure from my different project where we have email notifications set up on production errors. I'm changing the comment.